File: Python3/Easy/WrongAnswer/find-winner-on-a-tic-tac-toe-game_0.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        
        
        def check_win(board):
            for i in range(3):
                if board[i][0] == board[i][1] == board[i][2] != ' ':
                    return board[i][0]
                if board[0][i] == board[1][i] == board[2][i] != ' ':
                    return board[0][i]
            if board[0][0] == board[1][1] == board[2][2] != ' ':
                return board[0][0]
            if board[0][2] == board[1][1] == board[2][0] != ' ':
                return board[0][2]
            return None
        
        board = [[' ' for _ in range(3)] for _ in range(3)]
        for i, move in enumerate(moves):
            if i % 2 == 0:
                board[move[0]][move[1]] = 'X'
            else:
                board[move[0]][move[1]] = 'O'
            if i >= 4:
                winner = check_win(board)
                if winner:
                    return winner
        return 'Draw' if len(moves) == 9 else 'Pending'

File: Java/Easy/WrongAnswer/find-winner-on-a-tic-tac-toe-game_0.java
class Solution {
    public String tictactoe(int[][] moves) {
        
        int[][] board = new int[3][3];
        for (int i = 0; i < moves.length; i++) {
            if (i % 2 == 0) {
                board[moves[i][0]][moves[i][1]] = 1;
            } else {
                board[moves[i][0]][moves[i][1]] = 2;
            }
        }
        int sum = 0;
        for (int i = 0; i < 3; i++) {
            sum = board[i][0] + board[i][1] + board[i][2];
            if (sum == 3) {
                return "A";
            }
            if (sum == 6) {
                return "B";
            }
            sum = board[0][i] + board[1][i] + board[2][i];
            if (sum == 3) {
                return "A";
            }
            if (sum == 6) {
                return "B";
            }
        }
        sum = board[0][0] + board[1][1] + board[2][2];
        if (sum == 3) {
            return "A";
        }
        if (sum == 6) {
            return "B";
        }
        sum = board[2][0] + board[1][1] + board[0][2];
        if (sum == 3) {
            return "A";
        }
        if (sum == 6) {
            return "B";
        }
        if (moves.length == 9) {
            return "Draw";
        }
        return "Pending";
    }
}